test/ibmq/websocket/test_websocket_integration.py

```diff
@@ -60,19 +69,20 @@ def _job_final_status_polling(self, *args, **kwargs):
 
     def test_websockets_simulator(self):
         """Test checking status of a job via websockets for a simulator."""
-        job = self.sim_backend.run(self.bell, shots=1)
+        job = self._run_job(shots=1)
 
         # Manually disable the non-websocket polling.
         job._api_client._job_final_status_polling = self._job_final_status_polling
         result = job.result()
+        job.wait_for_final_state()
```

Comment:
nit - `job.result()` already waits for the job to reach a final state (hence the result can be returned).